design-docs/operationBased2.md

```diff
@@ -0,0 +1,269 @@
+data class WarmBloodedPetAnimal(val onAnimal: Animal.IOnAnimal): IAnimal
+// Fallback Model
+data class Animal(val onAnimal: IOnAnimal): IAnimal
+
+</pre></td>
+</tr>
+</table>
+
+
+## Example 7
+
+<table>
+<tr><th>GraphQL</th><th>Kotlin</th></tr>
+<tr>
+<td><pre lang="graphql">
+query TestOperation {
+  animal {
+    ... on Animal {
+      species
+      # Nested WarmBlooded
+      ... on WarmBlooded {
+        temperature
+      }
+      # Nested Pet
+      ... on Pet {
+        name
+      }
+    }
+  }
+}
+</pre></td>
+<td><pre lang="kotlin">
+
+// animal is not generated as polymorphic model because inline fragment ... on Animal is always assignable
+public data class Data(public val animal: Animal?)
+data class Animal(val onAnimal: IOnAnimal)
+
+// onAnimal field has two nested inline fragments -> generated as Polymorphic model
+public sealed interface IOnAnimal { 
+  val species: String,
+  val onWarmBlooded: OnAnimal.OnWarmBlooded?,
+   val onPet: OnAnimal.OnPet?
+}
+data class WarmBloodedOnAnimal(
+  val species: String, 
+  val onWarmBlooded: OnAnimal.OnWarmBlooded,
+  val onPet: OnAnimal.OnPet?
+): IOnAnimal
+data class PetOnAnimal(
+  val species: String,
+  val onWarmBlooded: OnAnimal.OnWarmBlooded?,
+  val onPet: OnAnimal.OnPet,
+): IOnAnimal
+data class WarmBloodedPetOnAnimal(
+  val species: String,
+  val onWarmBlooded: OnAnimal.OnWarmBlooded,
+  val onPet: OnAnimal.OnPet,
+): IOnAnimal
+data class OnAnimal(
+  val species: String,
+  val onWarmBlooded: OnWarmBlooded?,
+  val onPet: OnPet?,
+): IOnAnimal
+</pre></td>
+</tr>
+</table>
+
+## Example 8
+
+<table>
+<tr><th>GraphQL</th><th>Kotlin</th></tr>
+<tr>
+<td><pre lang="graphql">
+query TestOperation {
+  animal {
+    ... AnimalFragment
+  }
+}
+fragment AnimalFragment on Animal {
+  species
+  ... on WarmBlooded {
+    temperature
+  }
+  ... on Pet {
+    name
+  }
+}
+</pre></td>
+<td><pre lang="kotlin">
+
+// animal is not generated as polymorphic model because there is no inline fragment
+data class Animal(val animalFragment: IAnimalFragment)
+
+// AnimalFragment is generated as polymorphic model because it contains inline fragments
+sealed interface IAnimalFragment { 
+  val species: String
+  val onWarmBlooded: AnimalFragment.OnWarmBlooded?
+  val onPet: AnimalFragment.OnPet?
+}
+data class WarmBloodedAnimalFragment(
+  val species: String,
+  val onWarmBlooded: AnimalFragment.OnWarmBlooded,
+  val onPet: AnimalFragment.OnPet?
+): IAnimalFragment
+data class PetAnimalFragment(
+  val species: String,
+  val onWarmBlooded: AnimalFragment.OnWarmBlooded?,
+  val onPet: AnimalFragment.OnPet
+): IAnimalFragment
+data class WarmBloodedPetAnimalFragment(
+  val species: String,
+  val onWarmBlooded: AnimalFragment.OnWarmBlooded,
+  val onPet: AnimalFragment.OnPet
+): IAnimalFragment
+data class AnimalFragment(
+  val species: String,
+  val onWarmBlooded: AnimalFragment.OnWarmBlooded?,
+  val onPet: AnimalFragment.OnPet?
+): IAnimalFragment
```

Comment:
This is basically the same as example 7 with `OnAnimal` replaced with `AnimalFragment`?

```suggestion
public data class Data(public val animal: Animal?)
data class Animal(val animalFragment: AnimalFragment)

public sealed interface AnimalFragment { 
  val species: String,
  val onWarmBlooded: OnWarmBlooded?,
  val onPet: OnPet?
}
data class WarmBloodedAnimalFragment(
  val species: String, 
  val onWarmBlooded: OnWarmBlooded,
  val onPet:OnPet?
): AnimalFragment
data class PetAnimalFragment(
  val species: String,
  val onWarmBlooded: OnWarmBlooded?,
  val onPet: OnPet,
): AnimalFragment
data class WarmBloodedPetAnimalFragment(
  val species: String,
  val onWarmBlooded: OnWarmBlooded,
  val onPet: OnPet,
): AnimalFragment
data class OtherAnimalFragment(
  val species: String,
  val onWarmBlooded: OnWarmBlooded?,
  val onPet: OnPet?,
): AnimalFragment
```